controllers/clusterca_to_secret.go

```diff
@@ -0,0 +1,190 @@
+package controllers
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	Namespace  = "global-trust-certificates"
+	SecretName = "cluster-ca-secret"
+)
+
+func ClusterCaToSecret(client client.Client) {
+	log := log.Log.WithName("ClusterCAToSecret")
+
+	ticker := time.NewTicker(20 * time.Second)
+	quit := make(chan struct{})
+	for {
+		select {
+		case <-ticker.C:
+
+			isPresent, err := checkIfSecretPresent(context.Background(), client, log)
```

Comment:
You'll typically want re-use context instead of creating new with every call. In this case it probably won't matter but note that some code like the kubernetes client calls will use context to also listen for close events. The standard is usually one context per some cycle or fully served request. In this case I think one context per iteration of this for-loop.